compaction/compaction.cc

```diff
@@ -1419,6 +1427,7 @@ class scrub_compaction final : public regular_compaction {
     std::string _scrub_start_description;
     mutable std::string _scrub_finish_description;
     uint64_t _bucket_count = 0;
+    compaction_task_stats_ptr _cts = seastar::make_lw_shared<compaction_manager::task::compaction_task_stats>();
```

Comment:
Isn't a plain reference enough here?

Reply:
It is a `lw_shared_ptr` because of `make_scrubbing_reader` function: https://github.com/scylladb/scylla/blob/19b689e44fcc48f7a86144dbaacbc8e42b8d53a9/compaction/compaction.cc#L1493-L1496
It is used in a test only, so changing that would not be painful, but requires playing with the lifetime of a variable the value of which is not even needed for the test (to avoid dangling reference). What do you think?

Reply:
I think it is worth the change, but like I said I'm not sure we even need this `compaction_task_stats` anymore. As far as I understand the initial purpose of this was to be an output parameter to `compact_sstables()`, but I see that we now do that via `compaction_result` which is a much better way of doing this IMO.

Reply:
It's been changed to `uint64_t`. I will change that to reference.